### PR TITLE
[L0] Fix regular in order command list reuse given inorder queue

### PR DESCRIPTION
**Pull request by nrspruit**
- Fix command lists cache usage such that given an inorder queue and in order lists is requested, then only in order lists can be used from the cache.

### CODE REVIEW

File: source/adapters/level_zero/context.cpp [hunk 1]
```diff
@@ -712,6 +712,11 @@ ur_result_t ur_context_handle_t_::getAvailableCommandList(
 
     for (auto ZeCommandListIt = ZeCommandListCache.begin();
          ZeCommandListIt != ZeCommandListCache.end(); ++ZeCommandListIt) {
+      // If this is an InOrder Queue, then only allow lists which are in order.
+      if (Queue->Device->useDriverInOrderLists() && Queue->isInOrderQueue() &&
+          !(ZeCommandListIt->second.flags & ZE_COMMAND_QUEUE_FLAG_IN_ORDER)) {
```
Comment (raiyanla) [inline source/adapters/level_zero/context.cpp:717]: Should be ZE_COMMAND_LIST_FLAG_IN_ORDER ? Instead of QUEUE?

Comment (nrspruit) [reply source/adapters/level_zero/context.cpp]: fixed in the latest update.

**Review by raiyanla — APPROVED** [review]
LGTM now, assuming CI/tests all pass still.

**Comment by nrspruit** [comment]
This change is a blocker for https://github.com/oneapi-src/unified-runtime/pull/1370 which exposed this bug.

**Review by winstonzhang-intel — APPROVED** [review]

**Pull request merged**